bootloader/src/old_machine_common_functions.c

```diff
@@ -0,0 +1,109 @@
+#include <string.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include "old_machine_common_functions.h"
+
+char* getRequestDataJson(struct requestData reqData, char* requestFormat, char* systemStr){
+    char* tunnel;
+    if(reqData.tunnel != NULL){
+		size_t tunnelStringSize = sizeof(reqData.tunnel) + (2 * sizeof("\""));
+		tunnel = (char *)malloc(tunnelStringSize);
+        snprintf(tunnel, tunnelStringSize, "%s%s%s", "\"", reqData.tunnel, "\"");
+    } else {
+		tunnel = (char *)malloc(sizeof("false"));
+        strcpy_s(tunnel, sizeof("false"), "false");
+    }
+
+    size_t responseSize = strlen(requestFormat) + strlen(reqData.osVersion) + strlen(reqData.hostname) +
+					      strlen(tunnel) + strlen(reqData.IPstring) + strlen(systemStr);
+
+	if (reqData.glibcVersion != NULL) {
+		responseSize += strlen(reqData.glibcVersion);
+	}
+
+    // Concatenate into string for post data
+    char* buf = malloc(responseSize);
```

Comment:
You don't check malloc return value

Reply:
Either check snprintf return value or zero the memory after allocating it. To prevent cases where snprintf fails (for whatever reason) and we have junk data